src/main/java/edu/cornell/kfs/fp/document/service/impl/YearEndGeneralLedgerPendingEntriesServiceImpl.java

```diff
@@ -622,6 +662,97 @@ public GeneralLedgerPendingEntry generateBBCashOffset(AccountingDocumentBase doc
+         */
+        GeneralLedgerPendingEntry entry = new GeneralLedgerPendingEntry();
+        
+        // Populate a temporary OriginEntryFull object so that we can use the FlexibleOffsetAccountService functionality.
+        OriginEntryFull originEntry = new OriginEntryFull();
+        originEntry.setUniversityFiscalYear(new Integer(fiscalYear + 1));
+        originEntry.setChartOfAccountsCode(postable.getChartOfAccountsCode());
+        originEntry.setAccountNumber(postable.getAccountNumber());
+        originEntry.setSubAccountNumber(StringUtils.isBlank(postable.getSubAccountNumber())
+                ? KFSConstants.getDashSubAccountNumber() : postable.getSubAccountNumber());
+        originEntry.setFinancialObjectCode(postable.getFinancialObjectCode());
+        originEntry.setFinancialSubObjectCode(StringUtils.isBlank(postable.getFinancialSubObjectCode())
+                ? KFSConstants.getDashFinancialSubObjectCode() : postable.getFinancialSubObjectCode());
+        originEntry.setFinancialBalanceTypeCode(postable.getBalanceTypeCode());
+        originEntry.setFinancialObjectTypeCode(postable.getObjectCode().getFinancialObjectTypeCode());
+        // Use the FlexibleOffsetAccountService to make updates as needed.
+        try {
+            getFlexibleOffsetAccountService().updateOffset(originEntry);
+        } catch (InvalidFlexibleOffsetException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("generateCGBalanceForwardPendingEntry() Balance Forward Flexible Offset Error: " + e.getMessage());
+            }
+        }
+        
+        // Copy the potentially-changed data into the GL pending entry accordingly.
+        entry.setUniversityFiscalYear(originEntry.getUniversityFiscalYear());
+        entry.setChartOfAccountsCode(originEntry.getChartOfAccountsCode());
+        entry.setAccountNumber(originEntry.getAccountNumber());
+        entry.setSubAccountNumber(originEntry.getSubAccountNumber());
+        entry.setFinancialObjectCode(originEntry.getFinancialObjectCode());
+        entry.setFinancialSubObjectCode(originEntry.getFinancialSubObjectCode());
+        entry.setFinancialBalanceTypeCode(originEntry.getFinancialBalanceTypeCode());
+        entry.setFinancialObjectTypeCode(originEntry.getFinancialObjectTypeCode());
+        
+        // Continue with regular setup.
+        entry.setUniversityFiscalPeriodCode(KFSConstants.PERIOD_CODE_CG_BEGINNING_BALANCE);
+        entry.setFinancialDocumentTypeCode(document.getFinancialSystemDocumentHeader().getWorkflowDocument().getDocumentTypeName());
+        entry.setFinancialSystemOriginationCode(homeOriginationService.getHomeOrigination().getFinSystemHomeOriginationCode());
+        entry.setDocumentNumber(new StringBuilder(KFSConstants.BALANCE_TYPE_ACTUAL).append(postable.getAccountNumber()).toString());
+        entry.setTransactionLedgerEntrySequenceNumber(new Integer(sequenceNumber.intValue()));
```

Comment:
You're right, it shouldn't be necessary since Integer objects are immutable anyway. In this case, though, I was copying it to follow the setup of similar GL entry generation methods in the same class, which do use the intValue() call. I'll see if I can tweak this to not use the intValue() call.